theme/src/components/App/App.tsx

```diff
@@ -13,7 +19,30 @@ function InPageTableOfContents() {
 }
 
 function Content() {
-  const { pageTitle, pageBodyHtml } = useJupyterBookData();
+  const pageContentRef = useRef<HTMLDivElement | null>(null);
+  const router = useRouter();
+
+  const { globalHeaders, rootGlobalHeaders, pageTitle, pageBodyHtml } =
+    useJupyterBookData();
+
+  useEffect(() => {
+    const pageContentNode = pageContentRef.current;
+    if (!pageContentNode) {
+      return;
+    }
+
+    const linkNodes = Array.from(pageContentNode.querySelectorAll('a'));
```

Comment:
It looks like we're targeting all 'a' tags within the page content. What if there is an 'a' tag within the body content that isn't part of the TOC? Will it handle that properly?

For example, the "getting started" link on this page links to a separate page https://napari.org/tutorials/index.html

Reply:
While we get the list of all link nodes in the page content, we're only adding an event listener for internal links. The code here:

```ts
const href = link.href.replace(window.location.origin, '');
```

will convert any internal URLs to path names. So for the URL https://napari.org/tutorials/index.html, the value of `window.location.origin` will be `https://napari.org`, so `href` will result to `/tutorials/index.html`.

Reply:
Sorry, I was confused. I thought you were building the TOC from this list of links, nevermind!